check-symbiosis.lic

```diff
@@ -0,0 +1,80 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#check-symbiosis
+=end
+
+custom_require.call(%w[common common-arcana drinfomon spellmonitor])
+
+class SymbiosisCheck
+
+  def initialize
+    arg_definitions = [
+      [
+        {
+          name: 'symbiosis_type',
+          optional: true,
+          options: %w[activate avoid cast discern endure examine explore harness harvest heal impress learn perform remember resolve spell spring strengthen watch],
+          description: "Which symbiosis do you want to check? Argument -> Yaml -> defaults to harness"
+        }
+      ]
+    ]
+    args = DRC.parse_args(arg_definitions)
+
+    symbiosis_type = args.symbiosis_type || get_settings.symbiotic_research_type || 'harness'
+    validate_known_feat_for_symbiosis(symbiosis_type)
+    research_symbiosis(symbiosis_type)
+  end
+
+  def research_symbiosis(new_symbiosis)
+    current_symbiosis = DRCA.perc_symbiotic_research
+
+    # is the correct symbiotic research already up?
+    if current_symbiosis.downcase != new_symbiosis.downcase
+      # wrong symbiotic research is active, release it
+      if current_symbiosis
+        DRC.message("Wrong symbiotic research active, correcting")
+        DRC.message("Releasing: #{current_symbiosis}")
+        DRCA.release_magical_research
+      end
+      DRC.message("Researching: #{new_symbiosis}")
+      DRC.wait_for_script_to_complete('researcher', ['symbiosis', new_symbiosis])

```

Comment:
I wonder if we need to wait until that script to complete or simply start it and then let this script end naturally. 

If "check-symbiosis" doesn't need to still be running while "researcher" runs we could start the script via `start_script('researcher', ['symbiosis', new_symbiosis])`

The method "start_script" won't have a module prefix to it because that method is defined in core lich.rbw

It doesn't hurt to let this script wait, but it's also not necessary. 

Here's an example for reference, https://github.com/rpherbig/dr-scripts/blob/90f796710ed2ae3bc4812c182100df055ac92260/locksmithing.lic#L29

Reply:
Either method works, I was just more familiar with ``wait_for_script_to_complete``. But, since there is no reason to leave this running I'll switch.

Reply:
> I wonder if we need to wait until that script to complete or simply start it and then let this script end naturally.
> 
> If "check-symbiosis" doesn't need to still be running while "researcher" runs we could start the script via `start_script('researcher', ['symbiosis', new_symbiosis])`
> 
> The method "start_script" won't have a module prefix to it because that method is defined in core lich.rbw
> 
> It doesn't hurt to let this script wait, but it's also not necessary.
> 
> Here's an example for reference,
> 
> https://github.com/rpherbig/dr-scripts/blob/90f796710ed2ae3bc4812c182100df055ac92260/locksmithing.lic#L29

reverting the changes from this.. realized it breaks t2